msteams-platform/concepts/bots/bots-files.md

```diff
@@ -0,0 +1,215 @@
+    }
+  }]
+}
+```
+
+As a best practice, you should acknowledge the file upload by sending back a message to the user.
+
+## Uploading files to 1:1 chat
+Uploading a file to a user involves the following steps:
+1. Send a message to the user requesting permission to write the file. This message must contain a `FileConsentCard` attachment with the name of the file to be uploaded.
+2. If the user accepts the file download, your bot will receive an Invoke activity with a location URL in the `uploadInfo.uploadUrl` property.
+3. To transfer the file, your bot performs an HTTP PUT directly into the provided location URL.
+4. Optionally, you can perform a message update on the original message, removing the consent card if you do not want to allow the user to accept further uploads of the same file.
+
+### Message requesting permission to upload
+This message contains a simple attachment object requesting user permission to upload the file.
+
+![Screenshot of consent card requesting user permission to upload file](~/assets/images/bots/bot-file-consent-card.png)
+
+
+```json
+{
+  "attachments": [{
+    "contentType": "application/vnd.microsoft.teams.card.file.consent",
+    "name": "file_example.txt",
+    "content": {
+      "description": "<Purpose of the file, e.g. this is your monthly expense report>",
+      "sizeInBytes": 1029393,
+      "acceptContext": {        
+      },
+      "declineContext": {
+      }
+    }
+  }]
+}
+```
+
+The following table describes the content properties of the attachment:
+
+| Property | Purpose |
+| --- | --- |
+| `description` | Description of the file. May be shown to the user to describe its purpose or to summarize its content. |
+| `sizeInBytes` | Provides the user a hint as to the size of the file and the amount of space it will take in OneDrive. |
+| `acceptContext` | Additional context that will be silently transmitted to your bot when the user accepts the file. |
+| `declineContext` | Additional context that will be silently transmitted to your bot when the user declines the file. |
+
+### Invoke activity when the user accepts the file
+An invoke activity is sent to your bot if and when the user accepts the file. It contains the OneDrive for Business placeholder URL that your bot can then issue a PUT into to transfer the file contents. The following example shows an abridged version of the invoke activity that your bot will receive:
+
+```json
+{
+  ...
+
+  "name": "fileConsent/invoke",
+  "value": {
+    "type": "fileUpload",
+    "action": "accept",
+    "context": {
+    },
+    "uploadInfo": {
+      "contentUrl": "https://contoso.sharepoint.com/personal/johnadams_contoso_com/Documents/Applications/file_example.txt",
+      "name": "file_example.txt",
+      "uploadUrl": "https://upload.link",
+      "uniqueId": "1150D938-8870-4044-9F2C-5BBDEBA70C8C",
+      "fileType": "txt",
+    }
+  }
+}
+```
+
+Similarly, if the user declines the file, your bot will receive the following event:
+
+```json
+{
```

Comment:
Add "name" for consistency w/ accept example.

Reply:
Done